DEVELOPING.md

```diff
@@ -21,5 +21,58 @@ This will give you a minimal starting application, with a few basic basemaps, to
 either directly or by using the adminstration functions in the application to start adding layers, basemaps, versions and so on.
 
 ## Detailed guidelines
+First, clone the repositiry - the `main` branch should always be the latest, stable version.
 
-TODO
+### Setting up a database
+To run the project locally you will need to create a suitable database. We use Postgres but you can choose any database that is compatible with Entity Framework. If you’re using Postgres, install the PostGIS extension to enable geometries. 
+
+In your database, create a user for Entity Framework to use and give it permission to login and create databases.
+
+Create a schema within your database called giframeworkmaps. You may need to manually add the __EFMigrationsHistory table to the schema. To avoid complex permissions, make the user account you created the owner of the schema.
+
+### User secrets and connection strings in your IDE
+Once you’ve set up your database, add a user secret to your local copy of the application. In Visual Studio this is done by right-clicking on the web project and selecting “Manage user secrets”.
+
+In user secrets add the below line of code and fill in your database name, username and password:
+
+`{
+"ConnectionStrings:GIFrameworkMaps": "Host=localhost;Database=YourDatabaseName;Username=YourUserName;Password=YourPassword;SearchPath=giframeworkmaps"
+}`
+
+You should now be able to test your connection and run migrations using Entity Framework commands.
+
+### Download extensions in Visual Studio
+There are two extensions we use in Visual Studio that you can download: 
+- NPM Task Runner
+- EF Core Power Tools 
+
+You can download these by going to the Extensions menu and then to Manage Extensions. You’ll need to restart Visual Studio once you’ve downloaded these.
+
+### Authentication
+Follow these instructions if you want your app to have authentication so that only permitted users can access the administration part of the site.
+
+The Dorset Council version of this app uses Microsoft authentication, but you can use any provider that supports OpenID Connect. Refer to the documentation of your chosen provider for more information on how to setup an OpenID Connect connection.
+
+We use Azure KeyVault to provide the connection details to our production service. You can use any suitable service and free ones exist. Refer to the documentation of your chosen service on how to set it up.
```

Comment:
Not sure this is true. Its KeyVault, Environment Variables or App Config for all configuration items. We could have a section on how to add KeyVault, but this is outside the scope of this PR

Reply:
Wording amended to correct this